sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/implementation/ServiceBusReactorSession.java

```diff
@@ -88,6 +98,34 @@
         return createConsumer(linkName, entityPath, entityType, timeout, retry, receiveMode, filter);
     }
 
+    @Override
+    public Mono<AmqpLink> createSenderLink(String linkName, String entityPath, String viaEntityPath, Duration timeout,
+        AmqpRetryPolicy retry) {
+        Objects.requireNonNull(entityPath, "'entityPath' cannot be null.");
+        Objects.requireNonNull(timeout, "'timeout' cannot be null.");
+        Objects.requireNonNull(retry, "'retry' cannot be null.");
+
+        final Duration serverTimeout = adjustServerTimeout(timeout);
+        Map<Symbol, Object> linkProperties = new HashMap<>();
+
+        linkProperties.put(LINK_TIMEOUT_PROPERTY, UnsignedInteger.valueOf(serverTimeout.toMillis()));
+
+        if (!CoreUtils.isNullOrEmpty(viaEntityPath)) {
+            linkProperties.put(LINK_TRANSFER_DESTINATION_PROPERTY, entityPath);
+            logger.verbose("Get or create sender link for via entity path: '{}'", viaEntityPath);
+
+            final TokenManager tokenManager = tokenManagerProvider.getTokenManager(cbsNodeSupplier, entityPath);
```

Comment:
The token manager is Closeable. You never dispose of this resource. You don't need the check for endpoint states here the token manager already does it.